docs/development-setup.md

```diff
@@ -1,69 +1,147 @@
-
-Note: If `8.` fails with message `app-toolbelt: command not found`, you might have to set the build config differently first,
-e.g. to `nuernberg` and only then set the desired build config.
-
-Note: Your Intellij needs access to environment variables to run these commands successfully.
-
-As there are several services running and interacting some ports need to be forwarded.
+# Development Setup
+
+## Contents
+
+- [Prerequisites](#prerequisites)
+- [Frontend](#frontend)
+   - [Setup](#frontend-setup)
+   - [Run](#run-frontend)
+- [Backend](#backend)
+  - [Setup](#backend-setup)
+  - [Run](#run-backend)
+  - [Optional Setup](#optional-backend-setup)
+  - [Dumping and Restoring the Database](#dumping-and-restoring-the-database)
+  - [Using ehrenamtskarte.app as Database](#using-ehrenamtskarteapp-as-database)
+- [Administration](#administration)
+   - [Setup](#administration-setup)
+   - [Run](#run-administration)
+
+## Prerequisites
+
+1. Install the [protobuf compiler](https://github.com/protocolbuffers/protobuf) using a package manager of your choice
+  or using a binary from the [github release page](https://github.com/protocolbuffers/protobuf/releases). 
+  *The selected version has to match (or be older than) the version of protobuf-kotlin dependency used in [build.gradle.kts](../backend/build.gradle.kts).*
+2. Install Java JDK 17.
+   *If you use a later version, it has to be compatible with the [configured Gradle version](../frontend/android/gradle/wrapper/gradle-wrapper.properties).
+   Check the [Gradle Compatibility Matrix](https://docs.gradle.org/current/userguide/compatibility.html#java) for details.*
+3. Open the [root project](..) in IntelliJ.
+
+## Frontend
+
+### Frontend Setup
+
+1. Install Android SDK via the [Android plugin](https://www.jetbrains.com/help/idea/create-your-first-android-application.html#754fd) or Android Studio
+2. Install [fvm](https://fvm.app/documentation/getting-started/installation) (flutter version manager)
+3. Install flutter dependencies
+``` shell
+cd frontend && fvm flutter pub get
+```
+4. Open IntelliJ settings and
+   - Install the Android extension and set the Android SDK path
+   - Install the Dart extension in IntelliJ and set the Dart SDK path
+   - Install the Flutter extension in IntelliJ and set the Flutter SDK path
+5. Install app-toolbelt
+```shell
+npm install --unsafe-perm -g https://github.com/digitalfabrik/app-toolbelt/archive/refs/heads/main.tar.gz
+```
+
+*Note: IntelliJ needs access to environment variables to run these commands successfully.*
+
+### Run Frontend
+
+1. Forward ports. As there are several services running and interacting some ports need to be forwarded.
 This includes the backend port 8000 and the map tiles port at 5002.
-For this run:
+The command has to be run every time a device is connected.
+
+```shell
+adb reverse tcp:8081 tcp:8081 && adb reverse tcp:8000 tcp:8000 && adb reverse tcp:5002 tcp:5002
```

Comment:
This includes port forwarding for integreat as I figured having one command in the history for this makes sense. However, I am also open to keep it entitlementcard only.

Reply:
hm we have the `expose:ports` that can be triggered via `npm run expose:ports` i think you should even update this (it also checks for attached devices) and add the command here